packages/webapp/src/containers/SensorReadings/saga.js

```diff
@@ -43,157 +43,193 @@ import i18n from '../../locales/i18n';
+      currentUnixDate,
+      currentUnixTime,
+    } = getDates();
+    const formattedEndDate = moment(predictedUnixTime).format('MM-DD-YYYY');
 
     const header = getHeader(user_id, farm_id);
-    const endDate = moment(new Date().setDate(new Date().getDate() + 2)).format('MM-DD-YYYY');
     const postData = {
       farm_id,
       user_id,
       locationIds,
       readingTypes,
-      endDate: endDate,
+      endDate: formattedEndDate,
     };
-    const result = yield call(axios.post, sensorReadingsUrl(), postData, header);
-    const allSensorNames = result?.data?.sensorsPoints.map((s) => s.name);
-    const centerPoint = findCenter(result?.data?.sensorsPoints.map((s) => s?.point));
-    const activeReadingTypes = result?.data?.sensorsPoints[0].reading_type;
-    const currentDT = parseInt(+new Date() / 1000);
-
-    let lastUpdatedReadingsTime = {};
-    let xAxisLabel = {};
-    let isFound = false;
-    let predictedXAxisLabel = '';
-    let stationName = '';
-    let latestTemperatureReadings = {};
-    let selectedSensorName = '';
-    let ambientDataWithSensorsReadings = {};
-    let latestActualReadTimes = [];
-    let sensorReadingData = {};
-
-    for (let readingType of readingTypes) {
-      latestActualReadTimes = result?.data?.sensorReading[readingType]
-        .filter((cv) => (cv.value ? cv.value : cv.value === 0))
-        .map((cv) => new Date(cv.actual_read_time).valueOf() / 1000);
 
-      lastUpdatedReadingsTime[readingType] = getLastUpdatedTime(latestActualReadTimes);
+    const result = yield call(axios.post, sensorReadingsUrl(), postData, header);
+    const data = result?.data;
+    let sensorDataByLocationIds = {};
 
-      if (readingType === TEMPERATURE) {
-        params = {
-          ...params,
-          lat: centerPoint?.lat ?? lat,
-          lon: centerPoint?.lng ?? lng,
-        };
-        const openWeatherPromiseList = [];
-        for (const weatherURL of OPEN_WEATHER_API_URL_FOR_SENSORS) {
-          const openWeatherUrl = new URL(weatherURL);
-          for (const key in params) {
-            openWeatherUrl.searchParams.append(key, params[key]);
-          }
-          if (weatherURL === DAILY_FORECAST_API_URL) {
-            openWeatherUrl.searchParams.append('cnt', 1);
-          }
-          openWeatherPromiseList.push(call(axios.get, openWeatherUrl?.toString()));
-        }
+    for (let locationId of locationIds) {
+      sensorDataByLocationIds[locationId] = {};
+      for (let type of readingTypes) {
+        sensorDataByLocationIds[locationId][type] = {};
 
-        const [openWeatherResponse, predictedWeatherResponse, predictedDailyWeatherResponse] =
-          yield all(openWeatherPromiseList);
-        stationName = predictedDailyWeatherResponse?.data?.city.name;
-        const weatherResData = [
-          ...openWeatherResponse.data.list,
-          ...predictedWeatherResponse.data.list,
-        ];
+        let readings = sensorDataByLocationIds[locationId][type];
```

Comment:
I tried this and I liked it at first! 

But then I kind of liked having `sensorDataByLocationIds[locationId][type] = {};` 2 lines below `sensorDataByLocationIds[locationId] = {};` as an indicator that we are instantiating both objects. 

I could leave the line I like in and add your two lines instead of `let readings = sensorDataByLocationIds[locationId][type];` But then I am setting it twice... 

I don't know what is best there.